community-operators/enmasse/enmasse.0.28.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,974 @@
+          "metadata": {
+            "name": "myspace.user"
+          },
+          "spec": {
+            "authentication": {
+              "password": "ZW5tYXNzZQ==",
+              "type": "password"
+            },
+            "authorization": [
+              {
+                "addresses": [
+                  "myqueue"
+                ],
+                "operations": [
+                  "send",
+                  "recv"
+                ]
+              }
+            ],
+            "username": "user"
+          }
+        },
+        {
+          "apiVersion": "admin.enmasse.io/v1beta1",
+          "kind": "ConsoleService",
+          "metadata": {
+            "name": "console"
+          },
+          "spec": {}
+        },
+        {
+          "apiVersion": "iot.enmasse.io/v1alpha1",
+          "kind": "IoTConfig",
+          "metadata": {
+            "name": "default"
+          },
+          "spec": {}
+        },
+        {
+          "apiVersion": "enmasse.io/v1beta1",
+          "kind": "AddressSpaceSchema",
+          "metadata": {
+            "name": "undefined"
+          },
+          "spec": {}
+        }
+      ]
+
+spec:
+  maturity: beta
+  displayName: EnMasse
+  description: >
+    EnMasse provides a self-service messaging platform on Kubernetes and OpenShift with a uniform interface to manage different messaging infrastructure.
+    See our [website](http://enmasse.io) for more details about the project.
+
+    ### Supported features
+
+    * **Built-in authentication and authorization** - Use the built-in or plug in your own authentication service for authentication and authorization of messaging clients.
+
+    * **Self-service messaging for applications** - The service admin deploys and manages the messaging infrastructure, while applications can request messaging resources without caring about the messaging infrastructure.
+
+    * **Supports a wide variety of messaging patterns** - Choose between JMS-style messaging with strict guarantees, or messaging that supports a larger number of connections and higher throughput.
+
+    ### Configuring
+
+    The example resources provided this operator can be categorized into resources managed by the operator admin to configure the infrastructure and resources managed by messaging tenants to provision messaging.
+
+    1. Create the `AddressSpacePlan`, `AddressPlan`, `StandardInfraConfig`, `BrokeredInfraConfig` and `AuthenticationService` resources in the namespace where the operator is installed. See the [minimal service configuration](https://enmasse.io/documentation/master/kubernetes/index.html#minimal-service-configuration-messaging) for examples.
+    2. Create the `AddressSpace`, `Address` and `MessagingUser` resources in the same namespace as the application that will use messaging. See the [minimal tenant configuration](https://enmasse.io/documentation/master/kubernetes/index.html#minimal-tenant-configuration-messaging) for examples.
```

Comment:
Are these two "steps" really two different flows, the admin flow and the user flow?

Reply:
Yes, that would be an accurate way of describing it. We have defined the following personas (grouped into Admin and User):

* (Admin) Service Installer - Performs the initial installation, upgrades, and configuring plans
* (Admin) Service Operator - Monitors system level metrics (e.g. CPU / memory / storage) across all address spaces as part of the service offering
* (User) Tenant Admin - Owns the configuration of the address space, e.g. creates/deletes addresses, users, permissions
* (User) Developer - Creates applications which send/receive messages to an address space